mypy/nodes.py

```diff
@@ -2433,26 +2433,33 @@ class TypeVarLikeExpr(SymbolNode, Expression):
     Note that they are constructed by the semantic analyzer.
     """
 
-    __slots__ = ("_name", "_fullname", "upper_bound", "variance")
+    __slots__ = ("_name", "_fullname", "upper_bound", "default", "variance")
 
     _name: str
     _fullname: str
     # Upper bound: only subtypes of upper_bound are valid as values. By default
     # this is 'object', meaning no restriction.
     upper_bound: mypy.types.Type
+    default: mypy.types.Type
     # Variance of the type variable. Invariant is the default.
     # TypeVar(..., covariant=True) defines a covariant type variable.
     # TypeVar(..., contravariant=True) defines a contravariant type
     # variable.
     variance: int
 
     def __init__(
-        self, name: str, fullname: str, upper_bound: mypy.types.Type, variance: int = INVARIANT
+        self,
+        name: str,
+        fullname: str,
+        upper_bound: mypy.types.Type,
+        default: mypy.types.Type,
```

Comment:
Any reason not to make this default to `AnyType(TypeOfAny.from_omitted_generics)`? That would simplify some of the other code.

Reply:
AFAICT only simple literals are used as defaults in this file. I was thinking about `default: Type | None = None` but making it `None` complicates things a bit. It also makes sense not to set a default when looking at `upper_bound` which is often set to `self.object_type()`.